core/src/providers/llm.rs

```diff
@@ -493,12 +495,26 @@ impl LLMChatRequest {
                 );
                 Ok(c)
             }
-            Err(e) => Err(anyhow!(
-                "Error querying `{}:{}`: error={}",
-                self.provider_id.to_string(),
-                self.model_id,
-                e.to_string(),
-            )),
+            Err(e) => {
+                let string_error = e.to_string();
+                if let Ok(e) = e.downcast::<ModelError>() {
+                    //  Log the request_id if it exists.
+                    if let Some(request_id) = e.request_id {
+                        info!(
```

Comment:
`error!` ? Even if ~ similar for now might be better in the future?

Reply:
fixed. And improved the message as well

Reply:
And also added to the execute of regular llm request